target/linux/mvebu/files/arch/arm/boot/dts/armada-370-c200-v2.dts

```diff
@@ -0,0 +1,390 @@
+		};
+
+		disk2-red {
+			label = "red:disk2";
+			gpios = <&gpio1 19 GPIO_ACTIVE_LOW>;
+		};
+
+		disk-fail-green {
+			label = "green:disk-fail";
+			gpios = <&gpio1 23 GPIO_ACTIVE_LOW>;
+		};
+
+		disk-fail-red {
+			label = "red:disk-fail";
+			gpios = <&gpio1 21 GPIO_ACTIVE_LOW>;
+		};
+
+		usb1-green {
+			label = "green:usb1";
+			gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "usbport";
+			trigger-sources = <&usb1_port>;
+		};
+
+		usb1-red {
+			label = "red:usb1";
+			gpios = <&gpio0 15 GPIO_ACTIVE_LOW>;
+		};
+
+		usb2-green {
+			label = "green:usb2";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "usbport";
+			trigger-sources = <&usb2_port>;
+		};
+
+		usb2-red {
+			label = "red:usb2";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&coherencyfab {
+	broken-idle;
+};
+
+&eth1 {
+	pinctrl-0 = <&ge1_rgmii_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	phy-handle = <&ethphy0>;
+	phy-connection-type = "rgmii-id";
+};
+
+&i2c0 {
+	pinctrl-0 = <&i2c0_pins>;
+	pinctrl-names = "default";
+	clock-frequency = <100000>;
+	status = "okay";
+
+	nct7802@2a {
+		compatible = "nuvoton,nct7802";
+		reg = <0x2a>;
+	};
+
+	rtc@30 {
+		compatible = "s35390a";
+		reg = <0x30>;
+	};
+};
+
+&mdio {
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+
+	ethphy0: ethernet-phy@0 { /* Marvell 88E1318 */
+		reg = <0>;
+		#thermal-sensor-cells = <0>;
+	};
+};
+
+&nand_controller {
+	status = "okay";
+
+	nand@0 {
+		reg = <0>;
+		label = "pxa3xx_nand-0";
+		nand-rb = <0>;
+		marvell,nand-keep-config;
+		nand-on-flash-bbt;
+		nand-ecc-strength = <4>;
+		nand-ecc-step-size = <512>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "uboot";
+				reg = <0x0000000 0x200000>;
+			};
+
+			partition@200000 {
+				label = "certificate";
+				reg = <0x0200000 0x100000>;
+			};
+
+			partition@300000 {
+				label = "preset_cfg";
+				reg = <0x0300000 0x100000>;
+			};
+
+			partition@400000 {
+				label = "dev_params";
+				reg = <0x0400000 0x100000>;
+			};
+			partition@500000 {
+				label = "active_bank";
+				reg = <0x0500000 0x0100000>;
+			};
+
+			partition@600000 {
+				label = "magic";
+				reg = <0x0600000 0x0100000>;
+			};
+
+			partition@700000 {
+				label = "bank1";
+				reg = <0x0700000 0x2800000>;
+			};
+
+			partition@2f00000 {
+				label = "bank2";
+				reg = <0x2f00000 0x2800000>;
+			};
+
+			/* 0x5700000-0x5a00000 undefined in vendor firmware */
+
+			partition@5a00000 {
+				label = "reserved";
+				reg = <0x5a00000 0x2000000>;
+			};
+
+			partition@7a00000 {
+				label = "ubi";
+				reg = <0x7a00000 0x8600000>;
+			};
+		};
+	};
+};
+
+&pciec {
+	status = "okay";
+
+	pcie@1,0 {
+		pinctrl-0 = <&pmx_pcie>;
+		pinctrl-names = "default";
+		status = "okay";
+		reset-gpios = <&gpio1 27 GPIO_ACTIVE_LOW>;
+
+		/* -[0000:00]---01.0-[01]----00.0 */
+		/* usbport trigger won't work */
```

Comment:
@chunkeey 
Could You help me with figure out why Renesas uPD720202 led triger won't work?

Reply:
I set 4 triggers, but it won't help. I afraid, that reg addresses in pcie tree could be wrong. I'm not sure how to find it.

Reply:
> I afraid, that reg addresses in pcie tree could be wrong. I'm not sure how to find it.

hm, there's a quick test.

# cat /sys/class/leds/green:usb/trigger

if it's usbport then the regs are correct (and maybe port0+port1 are from the internal usb - though this wouldn't make sense?!). if not, then yeah.... this was a problem with the WNDR4700 + WNDAP660 too. they have a pcie-switch which resulted in a very strange paths.
I remember adding printk's into the kernel in order to get the correct lookup path (but sadly I no longer remember where exactly).

Reply:
addendum: 

a lot of the mvebu devices have upstream .dts. If this "looking for the correct regs" doesn't work out, you should posted your .dts to upstream ( Andrew Lunn <andrew@lunn.ch>, Gregory Clement <gregory.clement@bootlin.com>, Sebastian Hesselbarth <sebastian.hesselbarth@gmail.com>, linux-arm-kernel@lists.infradead.org ) and ask for their input.